.github/workflows/stm32build.yml

```diff
@@ -141,14 +141,20 @@ jobs:
             echo "latest=$(eval "./util/pcbversion.py STM32/${{ matrix.package }}/pcbversion latest both")" >> $GITHUB_OUTPUT
             echo "breaking=$(eval "./util/pcbversion.py STM32/${{ matrix.package }}/pcbversion breaking both")" >> $GITHUB_OUTPUT
           fi 
-
+      - name: Rename build files for Github release
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # upload only pushes to the main branch
+        # Rename .elf and .bin files to include release tag and pcb version in the Github release
+        run: |
+          RELEASENAME=${{ steps.release-on-push-2.outputs.tag_name }}-PCBv${{ steps.pcbversion.outputs.latest }}
+          cp STM32/${{ matrix.package }}/build/${{ matrix.package }}.elf ${RELEASENAME}.elf
+          cp STM32/${{ matrix.package }}/build/${{ matrix.package }}.bin ${RELEASENAME}.bin
```

Comment:
For my interest, did the {,} expansion not work? Or you thought it was too opaque?

Prefer using mv rather than cp, then you don't need to change the other lines.